ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController.swift

```diff
@@ -1972,12 +1972,20 @@ public class BrowserViewController: UIViewController {
       // Rewards reporting
       if let url = change?[.newKey] as? URL, !url.isLocal {
         // Notify Rewards of new page load.
-        if let rewardsURL = rewardsXHRLoadURL,
+        if let tab = tabManager.selectedTab,
+          let rewardsURL = rewardsXHRLoadURL,
           url.host == rewardsURL.host
         {
-          tabManager.selectedTab?.reportPageNavigation(to: rewards)
+          tab.reportPageNavigation(to: rewards)
+
+          rewards.reportTabUpdated(
```

Comment:
This needs to be discussed and re-thought because can possibly break site visits when navigating to the same eTLD+1